lib/dotiw/time_hash.rb

```diff
@@ -114,10 +114,17 @@ def build_years_months_weeks_days
         days_in_month = Time.days_in_month(last_month.month, last_month.year)
         weeks += days_in_month / 7
         days += days_in_month % 7
+
         if days >= 7
           days -= 7
           weeks += 1
         end
+
+        if weeks < 0
+          months -= 1
+          weeks += 5
+          days -= (7 - 3)
```

Comment:
`days` can equal `5` or `6`. In case of `5` it should be `1` and in case of `6` it should be `2`.